onyo/lib/onyo.py

```diff
@@ -576,6 +604,29 @@ def _init(self, directory: Union[Path, str]) -> None:
         from .command_utils import init_onyo
         return init_onyo(self, directory)
 
+    @staticmethod
+    def _is_onyo_root(directory: Path) -> bool:
+        """
+        Assert whether a directory is the root of a repository and has a fully
+        populated `.onyo/` directory.
+        """
+        dot_onyo = Path(directory, '.onyo')
+
+        if not dot_onyo.is_dir() or \
+                not Path(dot_onyo, "templates").is_dir() or \
+                not Path(dot_onyo, "validation").is_dir() or \
+                not Path(dot_onyo, "config").is_file() or \
+                not Path(dot_onyo, ".anchor").is_file() or \
+                not Path(dot_onyo, "templates/.anchor").is_file() or \
+                not Path(dot_onyo, "validation/.anchor").is_file():
```

Comment:
This can be shortened as:

```python3
dirs = ['templates', 'validation']
files = ['config', '.anchor', 'templates/.anchor', 'validation/.anchor']
has_dirs = [True if x.is_dir() else False for x in [dot_onyo / d for d in dirs]]
has_files = [True if x.is_file() else False for x in [dot_onyo / f for f in files]]

if any([*has_files, *has_dirs]):
    return False
```

Reply:
Well, if we are at it, then:

1. `True if x.is_dir() else False` is superfluous. `is_dir()` returns a bool, so `[x.is_dir() for x ....` it is ;-)
2. Here is what I find even nicer to read:

```python3
dirs = ['templates', 'validation']
files = ['config', '.anchor', 'templates/.anchor', 'validation/.anchor']
has_dirs = all(x.is_dir() for x in [dot_onyo / d for d in dirs])
has_files = all(x.is_file() for x in [dot_onyo / f for f in files])

if not (has_files and has_dirs):
    return False
```

Reply:
Of course I don't care much whether it's `not (has_files and has_dirs)` or `not has_files or not has_dirs`

Reply:
This is the kind of nitpicking I can get behind. Thanks for further improvements! 😄

Reply:
True.